backend-agent/app/db/utils.py

```diff
@@ -0,0 +1,71 @@
+from .models import (
+    Attack as AttackDB,
+    db,
+    AttackModel as AttackModelDB,
+    AttackResult as AttackResultDB,
+    ModelAttackScore as ModelAttackScoreDB,
+)
+
+
+def save_to_db(attack_results):
+    """
+    Persist the SuiteResult into the database.
+    Returns a list of AttackResults that were added.
+    """
+    inserted_records = []
+
+    attack_name = attack_results.attack.lower()
+    success = attack_results.success
+    vulnerability_type = attack_results.vulnerability_type.lower()
+    details = attack_results.details  # JSON column
+    model_name = details.get('target_model').lower() if 'target_model' in details else 'unknown'
+
+    model = AttackModelDB.query.filter_by(name=model_name).first()
+    if not model:
+        model = AttackModelDB(name=model_name)
+        db.session.add(model)
+        db.session.flush()
+
+    attack = AttackDB.query.filter_by(name=attack_name).first()
```

Comment:
what happens when we release a new attack? Is it added automatically with weight=1 or can we define a different weight somehow?

Reply:
Then please add a comment on this line not to forget. Even just something like
```
When a new attack is added, its weight is set to 1 by default. At this stage, the weight has to be changed manually in the db if 1 is not the correct value.
```